player.py

```diff
@@ -92,180 +93,153 @@ def toPlayers(playersString):
-                
-                
-                elif c == ")":
                     
-                    if not error and not change:
-                        try:
+                    elif c == ")":
+                        
+                        if not change:
                             currentExtract[3][-1] = float(currentExtract[3][-1])
-                        except:
-                            error = True
-                            currentExtract[3] = Player.BASE_SIZE
+                            change = True
                         
-                        change = True
+                        else:
+                            onUsername, onColor, onPosition, onSize = False, False, False, False
+                            change = False
                     
-                    if change:
-                        onUsername, onColor, onPosition, onSize = False, False, False, False
-                        change, error = False, False
-                
-                elif not error:
-                    if len(currentExtract[3]) == 0 or change:
-                        currentExtract[3].append(c)
-                        change = False
                     else:
-                        currentExtract[3][-1] += c
-            
-            
-            
-            # --------------- Create new player ---------------
-            else:
-                
-                if c == "[":
-                    pass
-                
-                elif c == "(":
-                    currentExtract = []
-                    onUsername = True
-                    onColor, onPosition, onSize = False, False, False
-                    change, error = False, False
+                        if len(currentExtract[3]) == 0 or change:
+                            currentExtract[3].append(c)
+                            change = False
+                        else:
+                            currentExtract[3][-1] += c
+                            
+                            
+                            
+                # --------------- Create new player ---------------
+                else:
+                    
+                    if c == "[":
+                        pass
+                    
+                    elif c == "(":
+                        currentExtract = []
+                        onUsername = True
+                        onColor, onPosition, onSize = False, False, False
+                        change = False
+                    
+                    elif c == "," or c == "]":
+                        playersList.append(Player(username=currentExtract[0],
+                                                color=tuple(currentExtract[1]),
+                                                position=tuple(currentExtract[2]),
+                                                size=tuple(currentExtract[3])))
+                        
+                        
+            except: # if any error occured
```

Comment:
Will a syntax error be detected as expected ?

Reply:
I tried a few different errors and they were all considered as such, but I may have forgotten one.
However if an error is not caught with this it means the program was able to create all players without problem, just the players may have values different than the expected ones but that's not something we can really verify.